[Vitest Integration] add vitest debugging guide (#20360)
cloudflare/cloudflare-docs: src/content/docs/workers/testing/vitest-integration/debugging.mdx

---
title: Debugging
pcx_content_type: concept
sidebar:
  order: 8
head: []
description: Debug your Workers tests with Vitest.
---

> Starting with `@cloudflare/vitest-pool-workers` v0.7.5, you can now debug your Workers tests with Vitest.

Reviewer: ```suggestion
This guide shows you how to debug your Workers tests with Vitest. This is available with `@cloudflare/vitest-pool-workers` v0.7.5 or later.
```
I find that writing it this way ages better over time.

## Open inspector with Vitest

To start debugging, run Vitest with the following command and attach a debugger to port `9229`:

```sh
vitest --inspect --no-file-parallelism
```

> ## Customizing the inspector port

Reviewer: ```suggestion
## Customize the inspector port
```
Keeps parallel structure with the other headings

By default, the inspector will be opened on port `9229`. If you need to use a different port (for example, `3456`), you can run the following command:

```sh
vitest --inspect=3456 --no-file-parallelism
```

> or define it in your Vitest configuration file:

Reviewer: ```suggestion
Alternatively, you can define it in your Vitest configuration file:
```

```ts
import { defineWorkersConfig } from "@cloudflare/vitest-pool-workers/config";

export default defineWorkersConfig({
  test: {
    inspector: {
        port: 3456,
    },
    poolOptions: {
      workers: {
        // ...
      },
    },
  },
});
```

## Setup VS Code to use breakpoints

> To setup VS Code for breakpoint debugging in your Worker tests, you can create a `.vscode/launch.json` file that contains the following configuration:

Reviewer: ```suggestion
To setup VS Code for breakpoint debugging in your Worker tests, create a `.vscode/launch.json` file that contains the following configuration:
```

```json
{
    "configurations": [
        {
            "type": "node",
            "request": "launch",
            "name": "Open inspector with Vitest",
            "program": "${workspaceRoot}/node_modules/vitest/vitest.mjs",
            "console": "integratedTerminal",
            "args": ["--inspect=9229", "--no-file-parallelism"]
        },
        {
            "name": "Attach to Workers Runtime",
            "type": "node",
            "request": "attach",
            "port": 9229,
            "cwd": "/",
            "resolveSourceMapLocations": null,
            "attachExistingChildren": false,
            "autoAttachChildProcesses": false,
        }
    ],
    "compounds": [
        {
            "name": "Debug Workers tests",
            "configurations": ["Open inspector with Vitest", "Attach to Workers Runtime"],
            "stopAll": true
        }
    ]
}
```

> You will find the **Debug Workers tests** option at the top of the **Run & Debug** panel. This will open inspector with Vitest and attach a debugger to the Workers runtime. You can then add breakpoints to your test files and start debugging.

Reviewer: ```suggestion
Select **Debug Workers tests** at the top of the **Run & Debug** panel to open inspector with Vitest and attach a debugger to the Workers runtime. Then you can add breakpoints to your test files and start debugging.
```
Should it be "an inspector with Vitest" or "**inspector with Vitest**". It seems like something is missing there?

Author: It should be `an inspector with Vitest`. I will fix that. Thanks for the review :)
